src/Raven.Server/Commercial/SetupManager.cs

```diff
@@ -0,0 +1,365 @@
+                        onProgress(progress);
+                        return null;
+                    }
+
+                    try
+                    {
+                        var csr = new CertificationRequestBuilder();
+                        csr.AddName($"CN={setupInfo.Domain}");
+                        csr.SubjectAlternativeNames.Add(setupInfo.Domain);
+                        
+                        // TODO make sure this is allowed, otherwise we need a challenge for every single node seperately
+                        foreach (var node in setupInfo.NodeSetupInfos)
+                        {
+                            csr.SubjectAlternativeNames.Add($"{node.NodeTag}.{setupInfo.Domain}");
+                        }
+                        var cert = await acmeClient.NewCertificate(csr);
+
+                        var pfxBuilder = cert.ToPfx();
+                        var certBytes = pfxBuilder.Build(setupInfo.Domain + " cert", "");
+                        var base64Cert = Convert.ToBase64String(certBytes);
+
+                        if (PlatformDetails.RunningOnPosix)
+                            AdminCertificatesHandler.ValidateCaExistsInOsStores(base64Cert, "Let's Encrypt certificate", _serverStore);
+                        
+
+                        // Prepare settings.json files, and write the local one to disk
+                        var settingsPath = SettingsFileName;
+                        var jsons = new Dictionary<string, string>();
+                        SecuredSetupInfo.NodeInfo localNode = null;
+
+                        foreach (var node in setupInfo.NodeSetupInfos)
+                        {
+                            try
+                            {
+                                if (node.NodeTag == "A")
+                                {
+                                    jsons.Add(node.NodeTag, WriteSettingsJsonFile(node.Certificate, node.PublicServerUrl, node.ServerUrl, settingsPath, SetupMode.LetsEncrypt, modifyLocalServer: true));
+                                    localNode = node;
+                                }
+                                else
+                                    jsons.Add(node.NodeTag, WriteSettingsJsonFile(node.Certificate, node.PublicServerUrl, node.ServerUrl, settingsPath, SetupMode.Secured, modifyLocalServer: false));
+                            }
+                            catch (Exception e)
+                            {
+                                throw new InvalidOperationException($"Failed to update {settingsPath} with new configuration.", e);
+                            }
+                        }
+
+                        // Need to return the jsons to the caller, so they can zip and send them to the studio
+
+                        try
+                        {
+                            var ips = localNode?.Ips.Select(ip => new IPEndPoint(IPAddress.Parse(ip), localNode.Port)).ToArray();
```

Comment:
Same issue as before, if there is a configuration problem, we want to remember the certificate is already there so we'll not regenerate it all the time.
Probably better to split this part entirely from the rest.
So we'll have the stage of generating the certificates and the zip file, then we'll validate the configuration, then we can proceeed.